ospd_openvas/daemon.py

```diff
@@ -1031,8 +1031,17 @@ def stop_scan_cleanup(
             # Set stop flag in redis
             kbdb.stop_scan(scan_id)
 
+            try:
+                ovas_process = psutil.Process(int(ovas_pid))
+            except psutil.NoSuchProcess:
+                ovas_process = None
```

Comment:
In case openvas is not running anymore this PID can be belong to any running process which happens to get the same PID as the old openvas process. The processes from psutil on the other hand will check if this process is the same, so it will guarantee that it is not performing actions on another process, which happens to have the same PID

Reply:
Yes, therefore I check for the Process name, to ensure that it is an "openvas" process

Reply:
I changed the signature to use the PID and not the Process, because in the case I am trying to handle with this PR, the orignal process doesn't exist (it is a new process with other PID), but what I want to terminate is the old one, still runnning. I have for that old scan just the PID.

Reply:
Ah you are right

Reply:
Fixed with 137f70e9793cf17d26b16c0102847d47423d8d61